### PR TITLE
[ENG-1132] Add Evaluation documentation pointing to Literal cookbooks

### PR DESCRIPTION
**Pull request by desaxce**
I added an explanation of Evaluation which points to the two cookbooks.

- - -
This change is [Reviewable](https://reviewable.io/reviews/Chainlit/literal-docs/25)

### CODE REVIEW

File: concepts/evaluation.mdx [hunk 1]
```diff
@@ -2,5 +2,62 @@
...
+building the most relevant KPIs for end-users:
+- Thread &mdash; user satisfaction
+- Agent Run &mdash; task completion
+- LLM Generation &mdash; hallucination
+
+An example is the vanilla Retrieval Augmented Generation (RAG), which augments Large Language Models
+(LLMs) with domain-specific data. Examples of metrics you can score against are:
+- context relevancy
+- faithfulness
+- toxicity
```
Comment (constantinidan) [inline concepts/evaluation.mdx:23]:
```suggestion
- context relevancy
- faithfulness
- answer relevancy
```

File: concepts/evaluation.mdx [hunk 2]
```diff
@@ -2,5 +2,62 @@
...
+versions perform better than previous ones. To best capture the **user experience** one must understand the
+multiple steps which make up the application. As AI applications grow in complexity, they tend to chain 
+multiple steps.
+
+Literal decorators let you catch the various steps of an AI application and gather the inputs
+and outputs of each. With those, one may evaluate the performance of their application at any level, 
+building the most relevant KPIs for end-users:
+- Thread &mdash; user satisfaction
+- Agent Run &mdash; task completion
+- LLM Generation &mdash; hallucination
```
Comment (constantinidan) [inline concepts/evaluation.mdx:17]:
```suggestion
Literal lets you log & monitor the various steps of your LLM application. By doing so, you can continuously improve the performance of your LLM system, building the most relevant metrics:
- Thread &mdash; user satisfaction, etc.
- Agent Run &mdash; task completion, etc.
- LLM Generation &mdash; hallucination, etc.
```

File: concepts/evaluation.mdx [hunk 3]
```diff
@@ -2,5 +2,62 @@
...
+- toxicity
+- etc.
+
+<Frame caption="Vanilla RAG Thread">
+  <img src="/images/rag-thread.png" alt="A thread on a RAG application" />
+</Frame>
+
+Literal offers a compelling user/developer interface to build and interact with datasets.
+Through `Dataset` objects, users can iterate on an application parameter and view
+its impact via Experiments.
```
Comment (constantinidan) [inline concepts/evaluation.mdx:32]:
```suggestion
Literal offers a compelling user/developer interface to build and interact with datasets.
Through `Dataset` objects, users can iterate on a parameter of the LLM system - such as prompt template, LLM provider, temperature, etc - and view its impact via Experiments. 
```

File: concepts/evaluation.mdx [hunk 4]
```diff
@@ -2,5 +2,62 @@
...
+<Frame caption="Vanilla RAG Thread">
+  <img src="/images/rag-thread.png" alt="A thread on a RAG application" />
+</Frame>
+
+Literal offers a compelling user/developer interface to build and interact with datasets.
+Through `Dataset` objects, users can iterate on an application parameter and view
+its impact via Experiments. 
+
+For instance, a developer can iterate on their prompt template and check the impact of their
+modifications on production data.
```
Comment (constantinidan) [inline concepts/evaluation.mdx:35]:
```suggestion
A developer can also iterate on their prompt template and check the impact of their
modifications on production data. 
```

File: concepts/evaluation.mdx [hunk 5]
```diff
@@ -2,5 +2,62 @@
...
+
+With offline evaluations, developers can leverage their favorite framework, such as
+promptfoo, Ragas, OpenAI Evals, LangChain Evaluators, etc.
+
+Our [Prompt Iteration with Promptfoo](https://github.com/Chainlit/literal-cookbook/tree/main/evaluation/typescript/prompt-iteration-promptfoo)
+cookbook shows how a developer can leverage the `promptfoo` library in TypeScript to
+compute similarity scores and persist the results of their experiments directly on Literal. 
+
+For Python aficionados, our [Context Relevancy with Ragas](https://github.com/Chainlit/literal-cookbook/tree/main/evaluation/python) notebook experiments on an example RAG
+application to score context relevancy.
```
Comment (constantinidan) [inline concepts/evaluation.mdx:52]:
```suggestion
- [Prompt Iteration with **Promptfoo**](https://github.com/Chainlit/literal-cookbook/tree/main/evaluation/typescript/prompt-iteration-promptfoo):
Our cookbook shows how a developer can leverage the `promptfoo` library in TypeScript to
compute similarity scores and persist the results of their experiments directly on Literal. 
- [Context Relevancy with **Ragas**](https://github.com/Chainlit/literal-cookbook/tree/main/evaluation/python): our notebook experiments on an example RAG
application to score context relevancy.
```

**Review by desaxce — COMMENTED** [review]
> Dismissed @constantinidan from a discussion.
> *[Reviewable](https://reviewable.io/reviews/chainlit/literal-docs/25)* status: 0 of 2 files reviewed, 4 unresolved discussions (waiting on @constantinidan)

___
*[`concepts/evaluation.mdx` line 17 at r1](https://reviewable.io/reviews/chainlit/literal-docs/25#gh-1566978924:-NvaGNhS7eua0Us07Xu9:b-896fix) ([raw file](https://github.com/chainlit/literal-docs/blob/0c0dd8c19d02e41bd4d86d0d5152b4e939d2a4b1/concepts/evaluation.mdx#L17)):*
<details><summary><i>Previously, constantinidan (DanConstantini) wrote…</i></summary><blockquote>

```suggestion
Literal lets you log & monitor the various steps of your LLM application. By doing so, you can continuously improve the performance of your LLM system, building the most relevant metrics:
- Thread &mdash; user satisfaction, etc.
- Agent Run &mdash; task completion, etc.
- LLM Generation &mdash; hallucination, etc.
```
</blockquote></details>

Done.
___
*[`concepts/evaluation.mdx` line 23 at r1](https://reviewable.io/reviews/chainlit/literal-docs/25#gh-1566972464:-NvaEsd-B5Z-eC6_wIzX:b-896fix) ([raw file](https://github.com/chainlit/literal-docs/blob/0c0dd8c19d02e41bd4d86d0d5152b4e939d2a4b1/concepts/evaluation.mdx#L23)):*
<details><summary><i>Previously, constantinidan (DanConstantini) wrote…</i></summary><blockquote>

```suggestion
- context relevancy
- faithfulness
- answer relevancy
```
</blockquote></details>

Done.
___
*[`concepts/evaluation.mdx` line 32 at r1](https://reviewable.io/reviews/chainlit/literal-docs/25#gh-1566986568:-NvaGaP2Bn-3V5PxxOmj:b-896fix) ([raw file](https://github.com/chainlit/literal-docs/blob/0c0dd8c19d02e41bd4d86d0d5152b4e939d2a4b1/concepts/evaluation.mdx#L32)):*
<details><summary><i>Previously, constantinidan (DanConstantini) wrote…</i></summary><blockquote>

```suggestion
Literal offers a compelling user/developer interface to build and interact with datasets.
Through `Dataset` objects, users can iterate on a parameter of the LLM system - such as prompt template, LLM provider, temperature, etc - and view its impact via Experiments. 
```
</blockquote></details>

Done.
___
*[`concepts/evaluation.mdx` line 35 at r1](https://reviewable.io/reviews/chainlit/literal-docs/25#gh-1566987146:-NvaGj7ICNrjI4phbmBr:b-896fix) ([raw file](https://github.com/chainlit/literal-docs/blob/0c0dd8c19d02e41bd4d86d0d5152b4e939d2a4b1/concepts/evaluation.mdx#L35)):*
<details><summary><i>Previously, constantinidan (DanConstantini) wrote…</i></summary><blockquote>

```suggestion
A developer can also iterate on their prompt template and check the impact of their
modifications on production data. 
```
</blockquote></details>

Done.
___
*[`concepts/evaluation.mdx` line 52 at r1](https://reviewable.io/reviews/chainlit/literal-docs/25#gh-1566990071:-NvaGt01A2baxw-rOIT5:b-896fix) ([raw file](https://github.com/chainlit/literal-docs/blob/0c0dd8c19d02e41bd4d86d0d5152b4e939d2a4b1/concepts/evaluation.mdx#L52)):*
<details><summary><i>Previously, constantinidan (DanConstantini) wrote…</i></summary><blockquote>

```suggestion
- [Prompt Iteration with **Promptfoo**](https://github.com/Chainlit/literal-cookbook/tree/main/evaluation/typescript/prompt-iteration-promptfoo):
Our cookbook shows how a developer can leverage the `promptfoo` library in TypeScript to
compute similarity scores and persist the results of their experiments directly on Literal. 
- [Context Relevancy with **Ragas**](https://github.com/Chainlit/literal-cookbook/tree/main/evaluation/python): our notebook experiments on an example RAG
application to score context relevancy.
```
</blockquote></details>

Done.

**Pull request merged**